src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/jvmsizing/underutilization/CpuUnderUtilizedRca.java

```diff
@@ -0,0 +1,126 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Resources.State;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.aggregators.SlidingWindow;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.aggregators.SlidingWindowData;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.contexts.ResourceContext;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.MetricFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.ResourceFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotResourceSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.ResourceUtil;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.RcaConf;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.scheduler.FlowUnitOperationArgWrapper;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.Record;
+import org.jooq.Result;
+
+public class CpuUnderUtilizedRca extends Rca<ResourceFlowUnit<HotResourceSummary>> {
+
+  private static final Logger LOG = LogManager.getLogger(CpuUnderUtilizedRca.class);
+  private static final long EVAL_INTERVAL_IN_S = 5;
+  private static final long DEFAULT_RCA_PERIOD = 60;
+  private static final double DEFAULT_UPPER_BOUND = 25D;
+  private final Metric cpuUtilization;
+  private final long rcaSamplesBeforeEval;
+  private final int numCores;
+  private long samples;
+  private double cpuUtilizationUpperBound;
+
+  private final SlidingWindow<SlidingWindowData> cpuUtilizationSlidingWindow;
+
+  public CpuUnderUtilizedRca(final Metric cpuUtilization) {
+    this(cpuUtilization, DEFAULT_RCA_PERIOD);
+  }
+
+  public CpuUnderUtilizedRca(final Metric cpuUtilization, final long rcaEvaluationIntervalInS) {
+    super(EVAL_INTERVAL_IN_S);
+    this.numCores = Runtime.getRuntime().availableProcessors();
+    this.cpuUtilization = cpuUtilization;
+    rcaSamplesBeforeEval = rcaEvaluationIntervalInS / EVAL_INTERVAL_IN_S;
+    this.cpuUtilizationSlidingWindow = new SlidingWindow<>((int) rcaEvaluationIntervalInS,
+        TimeUnit.MINUTES);
+    this.samples = 0;
+    this.cpuUtilizationUpperBound = DEFAULT_UPPER_BOUND;
+  }
+
+  @Override
+  public void generateFlowUnitListFromWire(FlowUnitOperationArgWrapper args) {
+    throw new UnsupportedOperationException("generateFlowUnitListFromWire should not be called "
+        + "for node-local rca: " + args.getNode().name());
+  }
+
+  @Override
+  public ResourceFlowUnit<HotResourceSummary> operate() {
+    samples++;
+    addToSlidingWindow();
+    if (samples == rcaSamplesBeforeEval) {
```

Comment:
You might want >= for these. It's unlikely, but rcaSamplesBeforeEval could technically be 0 and then you'd never evaluate